### PR TITLE
Added local notifications

### PR DESCRIPTION
**Pull request by TheHollidayInn**

### CODE REVIEW

File: SEDaily-IOS/NotificationsTableViewController.swift [hunk 1]
```diff
@@ -0,0 +1,200 @@
+//
+//  MoreTableViewController.swift
```
Comment (justinctlam) [inline SEDaily-IOS/NotificationsTableViewController.swift]: Comments needs to be updated

File: SEDaily-IOS/NotificationsTableViewController.swift [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    var customTabBarItem: UITabBarItem! {
+        return UITabBarItem(title: L10n.tabBarNotifications, image: #imageLiteral(resourceName: "mic_stand"), selectedImage: #imageLiteral(resourceName: "mic_stand_selected"))
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        let defaults = UserDefaults.standard
+        let subscribed = defaults.object(forKey: "sedaily-notificationsSubscribed") as? Bool
+        if subscribed != nil {
```
Comment (justinctlam) [inline SEDaily-IOS/NotificationsTableViewController.swift]: I think proper Swift would be if let subscribed = subscribed {}

File: SEDaily-IOS/NotificationsTableViewController.swift [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+
+    // MARK: - Table view data source
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        // #warning Incomplete implementation, return the number of sections
```
Comment (justinctlam) [inline SEDaily-IOS/NotificationsTableViewController.swift]: The warnings comment are unnecessary and can be removed

File: SEDaily-IOS/NotificationsTableViewController.swift [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        guard let cell = tableView.dequeueReusableCell(
+            withIdentifier: reuseIdentifier,
+            for: indexPath) as? NotificationTableViewCell else {
+                return UITableViewCell()
+        }
+        
+//        let cell = tableView.dequeueReusableCell(withIdentifier: reuseIdentifier, for: indexPath) as NotificationTableViewCell
```
Comment (justinctlam) [inline SEDaily-IOS/NotificationsTableViewController.swift]: Removed unused code

File: SEDaily-IOS/NotificationsTableViewController.swift [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+                print("Something went wrong")
+                return
+            }
+            
+            self.createNotification()
+        })
+    }
+    
+    func createNotification() {
+        let date = self.createDate(weekday: 2, hour: 10, minute: 0, year: 2018)
```
Comment (justinctlam) [inline SEDaily-IOS/NotificationsTableViewController.swift]: Why hardcode 2018, can you just use the user's current year?

File: SEDaily-IOS/NotificationsTableViewController.swift [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+        let request = UNNotificationRequest(identifier: "sedailyNotification", content: content, trigger: trigger)
+    
+        UNUserNotificationCenter.current().add(request) {(error) in
+            if let error = error {
+                print("Uh oh! We had an error: \(error)")
+            }
+        }
+    }
+
+    /*
```
Comment (justinctlam) [inline SEDaily-IOS/NotificationsTableViewController.swift]: unused code should be removed

File: SEDaily-IOS/NotificationsTableViewController.swift [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+            assignNotifications()
+            notificationsSubscribed = true
+        } else {
+            // cancel
+            UNUserNotificationCenter.current().removeAllPendingNotificationRequests()
+            notificationsSubscribed = false
+        }
+        
+        let defaults = UserDefaults.standard
+        defaults.set(notificationsSubscribed, forKey: "sedaily-notificationsSubscribed")
```
Comment (justinctlam) [inline SEDaily-IOS/NotificationsTableViewController.swift]: I would recommend storing the key in a constant for reuse

File: SEDaily-IOS/NotificationsTableViewController.swift [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+                return
+            }
+            
+            self.createNotification()
+        })
+    }
+    
+    func createNotification() {
+        let date = self.createDate(weekday: 2, hour: 10, minute: 0, year: 2018)
+        self.scheduleNotification(at: date, body: "We have new episodes for you!", titles: "Software Daily")
```
Comment (justinctlam) [inline SEDaily-IOS/NotificationsTableViewController.swift]: You may want to localize the body text

**Review by justinctlam — APPROVED** [review]

**Comment by TheHollidayInn** [comment]
Thanks for the review! I'm getting craig to help me with the layout library, then I'll post updates.

**Pull request merged**